dashboard/controller/assignments/index.js

```diff
@@ -24,15 +24,17 @@ exports.ini = function () {
 
 //main Loading page
 exports.index = function (req, res) {
-    console.log(req.session.user.user.role);
+    // reinit l10n and momemt with locale
     common.reinitLocale(req);
     var query = {
         sort: '+name'
     };
-
     //get query params
-    if (req.query.classroom != '') {
-        query['q'] = req.query.assignments;
+    if (req.query.assignment != '') {
+        query['q'] = req.query.assignment;
+    }
+    if (req.query.status != '') {
+        query['s'] = req.query.status;
```

Comment:
"q" is generally used for searching in name. Avoid using "q" and "s" for other parameters. Use full parameter name instead.